server/src/main/scala/actors/persistent/prediction/TouchdownPredictionActor.scala

```diff
@@ -0,0 +1,56 @@
+package actors.persistent.prediction
+
+import actors.persistent.staffing.GetState
+import actors.persistent.{RecoveryActorLike, Sizes}
+import uk.gov.homeoffice.drt.arrivals.VoyageNumber
+import uk.gov.homeoffice.drt.prediction.ModelAndFeatures
+import uk.gov.homeoffice.drt.time.SDateLike
+import drt.shared.CrunchApi.MillisSinceEpoch
+import org.slf4j.{Logger, LoggerFactory}
+import scalapb.GeneratedMessage
+import server.protobuf.messages.ModelAndFeatures.ModelAndFeaturesMessage
+import services.SDate
+import uk.gov.homeoffice.drt.ports.PortCode
+import uk.gov.homeoffice.drt.ports.Terminals.Terminal
+
+class TouchdownPredictionActor(val now: () => SDateLike,
+                               terminal: Terminal,
+                               number: VoyageNumber,
+                               origin: PortCode
+                              ) extends RecoveryActorLike {
+  import actors.serializers.ModelAndFeaturesConversion._
+
+  override val log: Logger = LoggerFactory.getLogger(getClass)
+
+  override val recoveryStartMillis: Long = now().millisSinceEpoch
+
+  var state: Option[ModelAndFeatures] = None
+
+  val uniqueId = s"${terminal.toString}-${number.numeric}-${origin.iata}"
+
+  override def persistenceId: String = s"touchdown-prediction-$uniqueId".toLowerCase
+
+  val sDateProvider: MillisSinceEpoch => SDateLike = (millis: MillisSinceEpoch) => SDate(millis)
+
+  override def processRecoveryMessage: PartialFunction[Any, Unit] = {
+    case msg: ModelAndFeaturesMessage =>
+      log.info(s"recovering state from ModelAndFeaturesMessage")
+      state = Option(modelAndFeaturesFromMessage(msg))
+  }
+
+  override def processSnapshotMessage: PartialFunction[Any, Unit] = {
+    case msg: ModelAndFeaturesMessage =>
+      state = Option(modelAndFeaturesFromMessage(msg))
+  }
+
+  override def stateToMessage: GeneratedMessage = throw new Exception(s"Persistence not supported here")
```

Comment:
just question , is this because we don't want snapshot to be persistent for this case ?

Reply:
The reason for this is that we're not persisting anything from this app. All persistence is done in the cron service from drt-analytics